File: leetcode/bit_manipulation/golang_solutions/power_of_two_231.go
package golang_solutions

func isPowerOfTwo(n int) bool {
	pow := 0
	num := 0
	for num < n {
		num = 1 << pow
		if num == n {
			return true
		}
		pow++
	}
	return false
}

func isPowerOfTwoBetter(n int) bool {
	return n != 0 && n&(n-1) == 0
}

File: python_solutions/math/easy/power_of_two_231.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        return n > 0 and (n & (n - 1)) == 0

    def isPowerOfTwoDevision(self, n: int) -> bool:
        if n <= 0:
            return False
        while n > 1:
            if n % 2 != 0:
                return False
            n = n // 2
        return True

    def isPowerOfTwoBruteForce(self, n: int) -> bool:
        if n == 1:
            return True
        elif n == 0:
            return False
        num = 2
        while True:
            if num == n:
                return True
            if num > n:
                return False
            num = num * 2
